test/e2e/network/dns_common.go

```diff
@@ -550,3 +549,51 @@ func validateTargetedProbeOutput(f *framework.Framework, pod *v1.Pod, fileNames
 
 	framework.Logf("DNS probes using %s succeeded\n", pod.Name)
 }
+
+// copied from https://golang.org/src/net/dnsclient.go
```

Comment:
The source copied from miekg/dns seems like it's based off an older Go version:

https://github.com/miekg/dns/blob/3b41a3134255be803026472cbb980adec17193c2/defaults.go#L328-L360

Reply:
Functionally they look the same but that one doesn't rely on `uitoa` interestingly.

Reply:
yeah, I noticed that but I've decided to take the more recent version

Reply:
is this everything below in this file?

can we maybe split it out and ensure correct license etc?
@kubernetes/dep-approvers the vendor review document doesn't really cover this.

I know in the past for larger chunks we've put code in `third_party/`.

Reply:
License is here https://github.com/kubernetes/kubernetes/blob/master/LICENSES/vendor/golang.org/x/net/LICENSE ,

Everything is there

If we have to do many things, I don't know if is worth the effort then Vs importing miekg :)

Reply:
The LICENSE doesn't apply to arbitrarily copying sources to other files AIUI, as it stands this file is misrepresented as all Kubernetes owned ASLv2, this utility should be clearly marked I think. I'm not sure exactly what we require for that, it might just be a distinct file with the appropriate headers.

Reply:
we still have one reference in pkg/proxy/winuserspace/proxysocket.go no?

Reply:
i'd lean towards picking up stuff directly from net/dnsclient.go and dropping it into third_party/ and fully removing someone's personal repo (and thus stick to the existing pattern)

Reply:
this comes with unit tests:
https://github.com/golang/go/blob/846dce9d05f19a1f53465e62a304dea21b99f910/src/net/lookup_test.go#L468-L504

> Functionally they look the same but that one doesn't rely on uitoa interestingly.

the strconv.AppendInt version is better organization wise, but the underlying formatBits has more branches and is likely a little heavier than uitoa

> i'd lean towards picking up stuff directly from net/dnsclient.go and dropping it into third_party/ and fully removing someone's personal repo (and thus stick to the existing pattern)

+1 for putting this under third_party/forked/golang

Reply:
it was simpler than I thought :)